test/Microsoft.AspNetCore.JsonPatch.Test/IntegrationTests/AnonymousObjectIntegrationTest.cs

```diff
@@ -0,0 +1,242 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Dynamic;
+using Microsoft.AspNetCore.JsonPatch.Exceptions;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.AspNetCore.JsonPatch.Test.IntegrationTests
+{
+    public class AnonymousObjectIntegrationTest
+    {
+        [Fact]
+        public void ShouldAddToList_WithDifferentCase()
+        {
+            var targetObject = new
+            {
+                IntegerList = new List<int>() { 1, 2, 3 }
+            };
+
+            // create patch
+            var patchDocument = new JsonPatchDocument();
+            patchDocument.Add("integerlist/0", 4);
+
+            var serialized = JsonConvert.SerializeObject(patchDocument);
+            var deserialized = JsonConvert.DeserializeObject<JsonPatchDocument>(serialized);
+
+            deserialized.ApplyTo(targetObject);
+
+            Assert.Equal(new List<int>() { 4, 1, 2, 3 }, targetObject.IntegerList);
+        }
+
+        [Fact]
+        public void AddToList_InvalidPositionTooLarge()
+        {
+            var targetObject = new
+            {
+                IntegerList = new List<int>() { 1, 2, 3 }
+            };
+
+            // create patch
+            var patchDocument = new JsonPatchDocument();
+            patchDocument.Add("IntegerList/4", 4);
+
+            var serialized = JsonConvert.SerializeObject(patchDocument);
+            var deserialized = JsonConvert.DeserializeObject<JsonPatchDocument>(serialized);
```

Comment:
We should blanket-remove all of these silliness. Add it to the list

Reply:
The fourth item is this type of clean up https://github.com/aspnet/JsonPatch/issues/105